### DISCUSSION TITLE
Brew for Cygwin; OpenBSD; FreeBSD; Darwin (BSD)?

### DISCUSSION THREAD

**Discussion by iamhumanipromise** [Homebrew/discussions#5141] [Everyday usage]
### Output of `brew config`

```shell
oh uhh...
```

### Output of `brew doctor`

```shell
just wanting some love...
```

### Description of issue

I'm a bit of a glutton for punishment. Windows is a necessity for many tasks; and cygwin takes away some of the pain. Its POSIX compliant (or nearly so) and should... SHOULD work fine with that. 

- Also; the BSDs could use a little bit of love: DragonBSD, OpenBSD, FreeBSD, NetBSD, GhostBSD, Mightnight BSD heck even the pfsense BSD
- GNU/Darwin for that BSD-like Unix-like Apple XNU kernely goodness? To let them all have packages! Yey packages! 
- PureDarwin for that downstream-delight! 
- Open UNIX-based Plan9-based OSs like Inferno or Harvey OS?
- Commercial UNIX System V Released 4-based AIX, Solaris or HP/UX? 
- Unix System V Release 4 descendant distros OpenIndiana, Tribblix, OmniOS, IllumOS (remnants of the OpenSolaris project)
- Homebrew for GNU/Hurd. (What other love does it even have?!) 

OK so yeah maybe the repos will be sparse and maybe I'm just pushing the limits of what a probably small, busy team can make happen until fans and lovers fill them repos UUUPPP. But we never know unless we try, you know? 

Homebrew is amazing and the platform truly brings cross-platform freedom. People out there have to love to tinker with new solutions on odd OSs and imagine the possibilities, no? 

A person can dream of the day homebrew becomes an OS-agnostic package manager. For real. 
Do you think Nix will end up fulfilling that role instead and I'm just misled?

**Comment by p-linnane** [comment]
I'm not 100% sure what you're asking here. Are you requesting official support for that bulleted list of systems? If so: it's very unlikely that we will be able invest our time as maintainers to target these systems. We officially support the 3 latest versions of macOS (same as Apple's support policy), as well as x86_64 Linux. Beyond that, it would require ensuring the `brew` package manger itself works on those systems, as well as an entire CI system to build bottles for each of those systems. That's unfortunately beyond the scope of what we can do as volunteers, especially given the usage numbers those systems would have.
